chainerx_cc/chainerx/array_repr.cc

```diff
@@ -257,7 +257,13 @@ std::ostream& operator<<(std::ostream& os, const Array& array) {
     // TODO(hvy): We need to determine the output specification of this function, whether or not to align with Python repr specification,
     // and also whether this functionality should be defined in C++ layer or Python layer.
     // TODO(hvy): Consider using a static dimensionality.
+
+#ifdef _WIN32
+	// hack for MSVC
+    VisitDtype(array.dtype(), [&os, &array](auto pt) { ArrayReprImpl<kDynamicNdim>{}.operator()<typename decltype(pt)::type>(array, os); });
```

Comment:
Doesn't this work in both conditions?
I think this version is better in the first place, because it's more explicit.

Reply:
Fixed it